jbpm-designer-client/src/main/resources/org/jbpm/designer/public/js/Plugins/inlinetaskformeditor.js

```diff
@@ -13,78 +13,207 @@ ORYX.Plugins.InlineTaskFormEditor = Clazz.extend({
 
     construct: function(facade){
         this.facade = facade;
-        this.facade.registerOnEvent(ORYX.CONFIG.EVENT_TASKFORM_EDIT, this.chooseFormEditor.bind(this));
+        this.facade.registerOnEvent(ORYX.CONFIG.EVENT_TASKFORM_EDIT, this.chooseFormEditorLoad.bind(this));
+        this.facade.registerOnEvent(ORYX.CONFIG.EVENT_TASKFORM_GENERATE, this.chooseFormEditorStore.bind(this));
+        this.facade.registerOnEvent(ORYX.CONFIG.EVENT_TASKFORM_GENERATE_ALL, this.chooseFormEditorStoreAll.bind(this));
     },
 
-    chooseFormEditor: function(options) {
-        Ext.Msg.show({
-            title : ORYX.I18N.inlineTaskFormEditor.formEditor,
-            msg : ORYX.I18N.inlineTaskFormEditor.selectForm,
-            buttons : {yes : ORYX.I18N.inlineTaskFormEditor.graphicalModeler, no : ORYX.I18N.inlineTaskFormEditor.markupEditor, cancel : ORYX.I18N.Dictionary.cancel},
-            icon : Ext.MessageBox.QUESTION,
-            fn : function(btn) {
-                if(btn == 'yes'){
-                    this.showTaskFormEditor("frm", options);
-                } else if(btn == 'no'){
-                    this.showTaskFormEditor("ftl", options);
+    chooseFormEditorLoad : function(options) {
+        this.chooseFormEditor(options, "load");
+    },
+
+    chooseFormEditorStore : function(options) {
+        this.chooseFormEditor(options, "store");
+    },
+
+    chooseFormEditorStoreAll : function(options) {
+        this.chooseFormEditor(options, "storeall");
+    },
+
+    chooseFormEditor: function(options, action) {
+        var editorAction = action;
+
+        if(ORYX.FORMSTYPE && (ORYX.FORMSTYPE == "form" || ORYX.FORMSTYPE == "frm")) {
+            if(action == "load") {
+                this.showTaskFormEditor(ORYX.FORMSTYPE, options);
+            } else if(action == "store") {
+                this.generateTaskForm(ORYX.FORMSTYPE, options);
+            } else if(action == "storeall") {
+                this.generateAllTaskForms(ORYX.FORMSTYPE, options);
+            } else {
+                this.facade.raiseEvent({
+                    type 		: ORYX.CONFIG.EVENT_NOTIFICATION_SHOW,
+                    ntype		: 'error',
+                    msg         : ORYX.I18N.inlineTaskFormEditor.errorInitiatingEditor+'.',
+                    title       : ''
+                });
+            }
+        } else {
+            Ext.Msg.show({
+                title : ORYX.I18N.inlineTaskFormEditor.formEditor,
+                msg : ORYX.I18N.inlineTaskFormEditor.selectForm,
+                buttons : {yes : ORYX.I18N.inlineTaskFormEditor.graphicalModeler, no : ORYX.I18N.inlineTaskFormEditor.graphicalModelerPreview, cancel : ORYX.I18N.Dictionary.cancel},
+                icon : Ext.MessageBox.QUESTION,
+                fn : function(btn) {
+                    if(btn == 'yes') {
```

Comment:
Hi Tihomir,

if I correct the difference between 'Yes' and 'No' branches are "form" and "frm" literals. I guess we can handl it in other way than code duplication. Move duplicated logic to separate function or use ternary operator:

```javascript
    var editorType = (btn == 'Yes') ? "form" : "frm";
    if (action == "load") {
         this.showTaskFormEditor(editorType, options);
    } ...........
```